OrbitLinuxTracing/UprobesReturnAddressManager.h

```diff
@@ -94,6 +86,16 @@ class UprobesReturnAddressManager {
       frames_to_patch.push_back(i);
     }
 
+    if (!tid_uprobes_stacks_.contains(tid)) {
+      // There are very rare cases where we don't have uprobes records, but need
```

Comment:
Do you have an idea of what might cause this? How does the callchain in this case usually like? Is it one uprobe frame or could there be more? Can such a frame be anywhere or for example only at the top? Could it be a sample inside uprobe code but right before the uprobe (not time-based) sample is collected?

Reply:
Not sure, why this happens. Frames look like, if we would process events out of order. But I don't think we do. 
So the frame I had was (given your example):
1us
10us
uprobe
100us

I only saw this with one uprobe for now. But I can easily check if this happens with others. Thats said, it is very rare (had like 0.01% to 0.02% of the samples). 
But it is not happening inside a uprobe.

Reply:
This is weird. So the u(ret)probe frame doesn't replace a frame but is actually in addition to the other ones?

Reply:
Either way be more descriptive in the comment about what is actually happening and also log an error if this doesn't happen too frequently.

Reply:
After more testing, I believe this results from lost and out-of-order events

Reply:
Are you using a debug build or are we losing events regularly?

Reply:
I adjusted the comment and added a log message

Reply:
Tested both. deb. and rel. build. I discovered the issue on debug build, but saw it once or twice (thus super rare) on release.

Reply:
Would you be able to test whether when you see this on a release build it's also due to losing events?

Reply:
Unsure, how. But I pretty much guess this is the case. The issue is basically the same as the other case (a bit below), were uprobes are missing.